pilot/pkg/bootstrap/server.go

```diff
@@ -196,6 +189,12 @@ func NewServer(args *PilotArgs) (*Server, error) {
 	}
 	s.initMeshNetworks(args, s.fileWatcher)
 
+	// Parse and validate Istiod Address.
+	host, port, err := e.GetDiscoveryAddress()
```

Comment:
This technically may not be the same. You could have a setup like

proxy -----> `gateway.istio.com` ----> `istiod.istio-system.svc`

Where you want the certs on istiod to be `istiod.istio-system.svc` but the address proxies connect to some gateway. But probably its fine for now